client/landing/stepper/README.md

```diff
@@ -1,106 +1,402 @@
 
-# Stepper Signup and Onboarding Framework
+# Stepper Walkthrough Framework
 
-The stepper framework is a new framework for quickly spinning up sign-up flows. This is a non-linear solution that doesn't sacrifice agility ✨.
+Stepper is a framework that allows you to make all kinds of walkthroughs. It's geared towards signup and onboarding flows, but you can use it to make any wizard.
 
-## Non-linearity
+## Table of Contents
 
-It has been tricky for us to create flows with input-driven steps configuration. Stepper makes it easy by using `useStepNavigation`. This hook has access to the state of the flow so it makes navigation decisions based on it at every stage of the flow.
+- [How does it work](#how-does-it-work)
+  - [The most important principle: Steps shouldn't make decisions or communicate](#the-most-important-principle-steps-shouldnt-make-decisions-or-communicate)
+- [Making a flow](#making-a-flow)
+  - [Code example](#code-example)
+  - [Registering the flow](#registering-the-flow)
+  - [File hierarchy convention](#file-hierarchy-convention)
+  - [Managing authentication](#managing-authentication)
+  - [Asserting conditions before running the flow](#asserting-conditions-before-running-the-flow)
+- [Notes](#notes)
+- [Deleting your flow](#deleting-your-flow)
+- [Making a Step](#making-a-step)
+  - [Code example](#code-example-1)
+  - [Passing data down to steps](#passing-data-down-to-steps)
+  - [Reusability](#reusability)
+  - [Renaming steps](#renaming-steps)
+- [State management](#state-management)
+  - [Typed state](#typed-state)
+  - [Miscellaneous fields](#miscellaneous-fields)
+- [Creating a site](#creating-a-site)
+- [The API](#the-api)
+- [Useful utilities and hooks](#useful-utilities-and-hooks)
+- [Troubleshooting](#troubleshooting)
+- [Help and feedback](#help-and-feedback)
 
-### Example flow
+## How does it work
+
+A Stepper flow **is not a series of steps**; it's a graph of steps with a non-deterministic order. The first step of the flow is deterministic, but the following steps are decided on the fly by the flow depending on state (user input, user status, site status, etc...).
+
+An ideal Stepper flow is a [finite-state machine](https://en.wikipedia.org/wiki/Finite-state_machine): it has a pre-defined set of possible states, and it should only be in one of those states at any given moment.
+
+### The most important principle: Steps shouldn't make decisions or communicate
+
+![StepperFlow](https://github.com/user-attachments/assets/c8d5b07a-4172-4ba1-92d0-7821f0b6b9f0)
+
+Stepper steps should be a lot like native form inputs. They may receive some properties, they may have internal state (think checkbox), and they must submit some information. That's it.
+
+1. They should not make narrative decisions (go back, go next, exit flow, etc..)
+2. They should not communicate with other steps (e.g. via query params).
+3. **They should not have side effects (e.g. persisting stuff in local storage).**
```

Comment:
They should not know in which flow they are being rendered, nor refer directly to any other steps.

I guess that any side effects that can affect the flow (such as writing query params in the URL) should ideally be avoided in the step, and performed in the flow itself